tools/lint-docstring

```diff
@@ -0,0 +1,162 @@
+
+
+def create_file_overview_doc() -> None:
+    """
+    Creates the entire developer-file-overview.md document again by retaining the
+    lines 1 to TABLE_OFFSET and the rest is written using the docstrings in the files
+    """
+    folder_file_docstring = generate_folder_file_docstrings_dict()
+    with open(DEVELOPER_DOC, "r+") as developer_file_read:
+        doc_data = developer_file_read.readlines()
+
+    files_and_docstrings_to_add = []
+    for folder in folder_file_docstring:
+        dictionary_of_files = folder_file_docstring[folder]
+
+        folder_text = folder
+        for file in sorted(dictionary_of_files):
+            docstring = folder_file_docstring[folder][file]
+            text_tuple = (folder_text, file, docstring)
+            folder_name, file_name, desc = column_text_with_spaces(
+                text_tuple, COLUMN_WIDTHS
+            )
+            new_line = f"| {folder_name}| {file_name}| {desc}|\n"
+            files_and_docstrings_to_add.append(new_line)
+            folder_text = " "
+
+        # adding blank row at the end of every folder
+        files_and_docstrings_to_add.append(BLANK_LINE)
+
+    # Folders that do not contain any files with docstrings are added separately to the file-overview
+    for skipped_folder_name in sorted(DOCSTRINGS_FOR_NO_FILE_FOLDERS):
+        skipped_text_tuple = (
+            skipped_folder_name,
+            " ",
+            DOCSTRINGS_FOR_NO_FILE_FOLDERS[skipped_folder_name],
+        )
+        folder_name, file_name, desc = column_text_with_spaces(
+            skipped_text_tuple, COLUMN_WIDTHS
+        )
+        new_line = f"| {folder_name}| {file_name}| {desc}|\n"
+        files_and_docstrings_to_add.append(new_line)
+        files_and_docstrings_to_add.append(BLANK_LINE)
+    doc_data[TABLE_OFFSET - 1 :] = files_and_docstrings_to_add[:-1]
+    updated_data = "".join(doc_data)
+    with open(DEVELOPER_DOC, "w") as developer_file_to_write:
+        developer_file_to_write.write(updated_data)
+
+
+def generate_folder_file_docstrings_dict() -> Dict[str, Dict[str, str]]:
+    """
+    Returns a dictionary containing folder name which in turn
+    is a dictionary containing files and their respective descriptions
+    """
+    total_files = extract_folder_file_structure()
+    folder_file_docstring: Dict[str, Dict[str, str]] = {}
+    for folder, files in sorted(total_files.items()):
+        folder_file_docstring[str(folder)] = {}
+        for file in files:
+            imported_file = importlib.import_module(
+                f'{folder.replace("/",".")}.{file[:-3]}'
+            )
+            extracted_docstring = str(imported_file.__doc__)
+            docstring = extracted_docstring.strip().replace("\n", " ")
+            if len(docstring) >= COLUMN_WIDTHS[2] + 1:
+                raise ValueError(
+                    f"{file} has docstring longer than {int(COLUMN_WIDTHS[2])+1}"
+                )
```

Comment:
Since space isn't included in the docstring, shouldn't we just say `"{file} has docstring longer than to {int(COLUMN_WIDTHS[2])}"`?
Similarily we can remove the = and +1 from the if condition?